dts/bindings/pinctrl/espressif,esp32-pinctrl.yaml

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Espressif's Pin controller Node
+    Based on pincfg-node.yaml binding.
+    Espressif's pin controller is in charge of controlling pin configurations, pin
+    functionalities and pin properties as defined by pin states. In its turn, pin
+    states are composed of pre-defined pin muxing definitions and user-provided
+    pin properties.
+
+    Each Zephyr-based application has its own set of pin muxing/pin configuration
+    requirements. The next  use ESP-WROVER-KIT's I2C_0 to illustrate how one could
+    change a node's pin state properties. Though based on a particular board, the
+    same following steps can be tweaked to address specifics of any other target
+    board.
+
+    Suppose an application running on top of the ESP-WROVER-KIT board, for some
+    reason it needs I2C_0's SDA signal to be routed to GPIO_33. When looking at
+    that board's original device tree source file (i.e., 'esp_wrover_kit.dts'),
+    you'll note that the I2C_0 node is already assigned to a set of states. Below
+    is highlighted the information that most interests us on that file
+
+
+        #include "esp_wrover_kit-pinctrl.dtsi"
+
+        &i2c0 {
+                pinctrl-0 = <&i2c0_sda_gpio21 &i2c0_scl_gpio22>;
+                pinctrl-names = "default";
+        };
+
+
+    From the above excerpt, the pincrl-0 property is composed of 'i2c0_sda_gpio21'
+    and 'i2c0_scl_gpio22' pin states. The naming convention makes it clear that
+    I2C_0's SDA signal is, by default, routed to GPIO_21 on the target board. Pin
+    states are defined on another file (i.e, 'esp_wrover_kit-pinctrl.dtsi') on
+    the same folder of the DTS file. Check below the excerpt describing SDA's
+    pin state
+
+
+        i2c0_sda_gpio21: i2c0_sda_gpio21 {
+                pinmux = <I2C0_SDA_GPIO21>;
+                bias-pull-up;
+                drive-open-drain;
+                output-high;
```

Comment:
nit: is this still correct (ie i2c pull-up)?

Reply:
The difference now is that - differently from the previous version of this PR - using this and other properties is no longer mandatory, but the option was kept in case one's wants to use SoC's internal pull-up at its own discretion.

Reply:
in general internall pull-ups are not suitable for I2C (even though in many cases "it just works" for certain configurations).